### PR TITLE
feat: [CDS-71283]: implement ServerlessPrepareRollback step

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Implement edit and runtime view of ServerlessPrepareRollback step
- Refactor existing component for common fields that was used in AwsSam steps
- Newly created common component will be used across AwsSam and Serverless steps

#### Screenshots
Screenshot 2023-06-16 at 1 43 32 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        <Container className={cx(stepCss.formGroup)}>
+          <MultiTypeList
+            name={commandOptionsFieldName} // 'spec.buildCommandOptions' / 'spec.deployCommandOptions'
+            multiTextInputProps={{
+              expressions,
+              allowableTypes: [MultiTypeInputType.FIXED, MultiTypeInputType.EXPRESSION]
+            }}
+            multiTypeFieldSelectorProps={{
+              label: getString('optionalField', {
+                name: commandOptionsFieldLabel // getString('cd.steps.awsSamBuildStep.awsSamBuildCommandOptions') / getString('cd.steps.awsSamDeployStep.awsSamDeployCommandOptions')
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx:153]: Is this commented code needed in future?

File: src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+            allowableTypes,
+            disabled: readonly
+          }}
+          tooltipProps={{ dataTooltipId: 'privileged' }}
+          disabled={readonly}
+          configureOptionsProps={{ hideExecutionTimeField: true }}
+        />
+      </Container>
+
+      <Container className={stepCss.formGroup}>
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx:215]: Do we need multiple `<Container` elements?

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    connectorRef: ConnectorRef
+    image?: string
+    serverlessVersion?: string
+    privileged?: boolean
+    imagePullPolicy?: string
+    runAsUser?: string
+    resources?: {
+      limits?: {
+        memory?: string
+        cpu?: string
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:55]: all optional?

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    }
+  }
+}
+export interface ServerlessPrepareRollbackStepProps {
+  initialValues: ServerlessPrepareRollbackStepInitialValues
+  onUpdate?: (data: ServerlessPrepareRollbackStepFormikValues) => void
+  stepViewType?: StepViewType
+  allowableTypes: AllowedTypes
+  readonly?: boolean
+  isNewStep?: boolean
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:66]: Do we have all of these optional?

**Review by ishantmehtaharness — APPROVED** [review]
Few comments, Rest looks good to me

File: src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx (hunk 2 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx:215]: Yup, needed for proper layout

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx (hunk 3 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:55]: Yes, this is as per BE contracts.

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx (hunk 4 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:66]: Yes, need to keep them optional as they are optional in StepProps itself.

File: src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx]: Removed.

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Comment by chetanvekariya-harness** [comment]
retrigger coverage

File: src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/Common/AwsSamServerlessStepCommonOptionalFields/AwsSamServerlessStepCommonOptionalFieldsEdit.tsx]: cool

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:55]: Is it possible to make the necessary ones required by BE and FE both, in follow up PR's?

File: src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx (hunk 4 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessAwsLambda/ServerlessPrepareRollbackStep/ServerlessPrepareRollbackStepEdit.tsx:66]: I see. so this happens for all components

**Review by ishantmehtaharness — APPROVED** [review]
PR looks good to me

**Review by vikrant-harness — APPROVED** [review]

**Pull request merged**